In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_csv("/content/imdb_data.csv")

In [ ]:
#Lets check whether the data is load or not. check first 5 rows

In [3]:
df.head()

id    primaryTitle   originalTitle  isAdult  runtimeMinutes  \
0  tt0035423  Kate & Leopold  Kate & Leopold        0             118   
1  tt0065421  The Aristocats  The AristoCats        0              78   
2  tt0065938  Kelly's Heroes  Kelly's Heroes        0             144   
3  tt0066026         M*A*S*H         M*A*S*H        0             116   
4  tt0066206          Patton          Patton        0             172   

                       genres  averageRating  numVotes    budget       gross  \
0      Comedy,Fantasy,Romance            6.4     87925  48000000  76019048.0   
1  Adventure,Animation,Comedy            7.1    111758   4000000  35459543.0   
2        Adventure,Comedy,War            7.6     52628   4000000   5200000.0   
3            Comedy,Drama,War            7.4     75784   3500000  81600000.0   
4         Biography,Drama,War            7.9    106476  12000000  61749765.0   

        release_date              directors  
0  December 11, 2001          James Mangold  
1  December 11, 1970    Wolfgang Reitherman  
2    January 1, 1970        Brian G. Hutton  
3   January 25, 1970          Robert Altman  
4   February 4, 1970  Franklin J. Schaffner

In [4]:
#basic information about dataset.

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3348 entries, 0 to 3347
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   id              3348 non-null   object 
 1   primaryTitle    3348 non-null   object 
 2   originalTitle   3348 non-null   object 
 3   isAdult         3348 non-null   int64  
 4   runtimeMinutes  3348 non-null   int64  
 5   genres          3348 non-null   object 
 6   averageRating   3348 non-null   float64
 7   numVotes        3348 non-null   int64  
 8   budget          3348 non-null   int64  
 9   gross           3297 non-null   float64
 10  release_date    3343 non-null   object 
 11  directors       3348 non-null   object 
dtypes: float64(2), int64(4), object(6)
memory usage: 314.0+ KB


- The dataset contain 3348 rows and 12 columns

- There are 6 object data type . 4 integer columns, 2 float columns.

- isAdult column showing int. we need to convert this columns into Yes and No format to analyse the data properly.

- release_date column is an object , which should be converted to date time format or  we seperate the date, year and month for better analysis.

In [6]:
#lets check for the missing values.


In [7]:
df.isnull().sum()

id                 0
primaryTitle       0
originalTitle      0
isAdult            0
runtimeMinutes     0
genres             0
averageRating      0
numVotes           0
budget             0
gross             51
release_date       5
directors          0
dtype: int64

- gross (51 missing values )

- gross represent total earning of a movies. missing values may indicate that data was unavailable or the movie is underperformed.

-possible solution - 1 -missing values replace with mean or median

2 - drop rows with missing values - 51 rows will be droped from the dataset

3 - If budget is available . then we estimate the gross using a median.


- missing release dates - incomplete record.
- possible solution rop the rows.

In [8]:
#lets replace the missing values of gross with KNN imputation

In [9]:
from sklearn.impute import KNNImputer

In [10]:
imputer = KNNImputer(n_neighbors=5)

In [11]:
df[['budget', 'gross']] = imputer.fit_transform(df[['budget', 'gross']])

In [12]:
df.isnull().sum()

id                0
primaryTitle      0
originalTitle     0
isAdult           0
runtimeMinutes    0
genres            0
averageRating     0
numVotes          0
budget            0
gross             0
release_date      5
directors         0
dtype: int64

In [13]:
#drop release date rows

In [14]:
df = df.dropna(subset=['release_date'])

In [15]:
df.shape

(3343, 12)

- After successfully removing the null values we have clean dataset.

- total 3343 rows and 12 columns.

- so lets start to do EDA

In [16]:
df.columns

Index(['id', 'primaryTitle', 'originalTitle', 'isAdult', 'runtimeMinutes',
       'genres', 'averageRating', 'numVotes', 'budget', 'gross',
       'release_date', 'directors'],
      dtype='object')